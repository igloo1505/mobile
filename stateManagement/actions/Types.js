export const SET_LOADING = "SET_LOADING";
export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOGIN_FAIL = "LOGIN_FAIL";
export const REGISTER_SUCCESS = "REGISTER_SUCCESS";
export const REGISTER_FAIL = "REGISTER_FAIL";
export const ADD_MEDICATION = "ADD_MEDICATION";
export const MEDICATION_TAKEN = "MEDICATION_TAKEN";
export const MEDIATION_ADJUST_TIMING = "MEDICATION_ADJUST_TIMING";
export const MEDICATION_REMOVE = "MEDICATION_REMOVE";
export const MEDICATION_ERROR = "MEDICATION_ERROR";
export const LOGOUT_USER = "LOGOUT_USER"
export const SET_FOCUSED = "SET_FOCUSED"
export const TOGGLE_ADD_TIMER_MODAL = "TOGGLE_ADD_TIMER_MODAL"