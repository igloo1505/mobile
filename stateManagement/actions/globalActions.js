import {SET_FOCUSED} from "./Types"

export const setFocused = (focusScreen) => async (dispatch) => {
    dispatch({ type: SET_FOCUSED, payload: focusScreen})
}