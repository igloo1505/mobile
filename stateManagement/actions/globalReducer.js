import { SET_LOADING, SET_FOCUSED } from "./Types";
import { AsyncStorage } from "react-native";
import { Alert } from "react-native";

const initialState = {
  triedAutoLogin: false,
  loading: false,
  focused: "",
  error: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_FOCUSED:
      return {
        ...state,
        focused: action.payload
      }
    default:
      return state;
  }
};
