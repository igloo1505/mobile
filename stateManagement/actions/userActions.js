import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_USER,
  REGISTER_SUCCESS,
  REGISTER_FAIL,
  SET_LOADING,
} from "./Types";
import Axios from "axios";
import { AppConstants } from "../../constants/appLevelConstants";

const { serverRoot } = AppConstants;
const config = {
  headers: {
    "Content-Type": "application/json",
  },
};

export const loginUser = (user) => async (dispatch) => {
  try {
    console.log("Running loginUser...");
    const res = await Axios.post(`${serverRoot}/authenticate`, user, config);
    // console.log("res: ", res);
    dispatch({ type: LOGIN_SUCCESS, payload: res.data });
  } catch (error) {
    console.log("Caught Error:", error);
    dispatch({ type: LOGIN_FAIL, payload: error.response });
  }
};

export const registerUser = (user) => async (dispatch) => {
  try {
    console.log("Running register User...")
      const res = await Axios.post(`${serverRoot}/users`, user, config)
      dispatch({ type: REGISTER_SUCCESS, payload: res.data})
      console.log("res.data: ", res.data)
  } catch (error) {
    console.log("Caught Error:", error);
    dispatch({ type: REGISTER_FAIL, payload: error.response });
  }
}


export const logoutUser = () => async (dispatch) =>{
  console.log("running Logout User...")
  dispatch({ type: LOGOUT_USER})
}

