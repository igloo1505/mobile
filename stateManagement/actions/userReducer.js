import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_USER,
  REGISTER_SUCCESS,
  REGISTER_FAIL,
} from "./Types";
import setAuthToken from "../setAuth";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Alert } from "react-native";


const getToken = async () => {
  let token = await AsyncStorage.getItem("token")
  if(token){
    return token
  }
  else return null
}

const initialState = {
  loggedIn: true,
  token: null,
  user: null,
  triedAutoLogin: false,
  error: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      setAuthToken(action.payload.token);
      AsyncStorage.setItem("token", action.payload.token);
      return {
        ...state,
        user: action.payload.user,
        loggedIn: true,
      };
      case LOGOUT_USER: 
      console.log("running logout user from useDispatch")
      AsyncStorage.removeItem("token")
      return {
        ...state,
        loggedIn: false,
        token: null,
        user: null,
        triedAutoLogin: false,
        error: null
      }
    default:
      return state;
  }
};
