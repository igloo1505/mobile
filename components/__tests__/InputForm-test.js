import React from "react";
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import TestRenderer from "react-test-renderer";
import {LoginForm} from "../LoginForm";
import {loginUser} from '../../stateManagement/actions/userActions'

const mockProps = { 
  toggleLogin: () => console.log('login Toggled'),
  loginUser: () => loginUser()
}

it("renders loginForm correctly", () => {

  const tree = TestRenderer.create(<LoginForm {...mockProps}  />).toJSON();
  expect(tree).toMatchSnapshot()
})

const TestLogin = async (user) => {
  const res = await loginUser(user)
  return res;
}

test('Test correct login', async () => {
  const User = { email: "aiglinski@icloud.com",
  password: "SomePassword123" };
  // const data = await TestLogin(User);
  await expect(TestLogin(User)).resolves.toBe('Andrew Iglinski');
});