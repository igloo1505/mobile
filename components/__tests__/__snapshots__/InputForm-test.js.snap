// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loginForm correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignContent": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    accessible={true}
    contentContainerStyle={
      Object {
        "alignContent": "center",
        "flex": 1,
      }
    }
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "margin": 20,
              "padding": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.26,
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Email
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <Text />
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                editable={true}
                id="emailTest"
                keyboardType="default"
                onChangeText={[Function]}
                placeholder="Email"
                placeholderTextColor="#86939e"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                textContentType="username"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#86939e",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Password
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <Text />
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                id="passwordTest"
                minLength={8}
                onChangeText={[Function]}
                placeholder="Password"
                placeholderTextColor="#86939e"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                textContentType="password"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "margin": 5,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "color": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                }
              }
            >
              Login
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            forwardedRef={[Function]}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderBottomWidth": 1,
                "borderColor": "#007aff",
                "borderLeftWidth": 1,
                "borderRadius": 5,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
                "borderWidth": 1,
                "color": undefined,
                "elevation": null,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginTop": 5,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 16.5,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
